src/library/scala/collection/immutable/Vector.scala

```diff
@@ -1200,6 +1227,21 @@ private[immutable] trait VectorPointer[+T] {
     }
 
 
+  /**
+    *
+    * @param array
+    * @return
+    */
+    @`inline` private def safeClone(array: Array[AnyRef]): Array[AnyRef] = {
+      val x = new Array[AnyRef](array.length)
+      var i = 0
+      while (i < array.length) {
+        x(i) = array(i)
+        i += 1
+      }
+      x
```

Comment:
should probably `System.arraycopy`

Reply:
`java.util.Arrays.copyOf` can do it without redundant zeroing of the new array

Reply:
We can't do `j.u.Arrays.copyOf` because we need the result to be an `Array[AnyRef]`, but
```scala
scala> java.util.Arrays.copyOf(Array(""), 1).getClass.getComponentType
res1: Class[_] = class java.lang.String
```

Reply:
Oh whoops, didn't see the other overload of `copyOf`, ok will make that change. 

```scala
scala> java.util.Arrays.copyOf[AnyRef,AnyRef](Array(""), 1, classOf[Array[AnyRef]])
res4: Array[AnyRef] = Array("")
```